lib/pages/general/item/item.dart

```diff
@@ -491,8 +524,94 @@ class ItemPage extends StatelessWidget {
         return 'Crafting Material';
       case 'UpgradeComponent':
         return 'Upgrade Component';
+
+      case 'AccountBindOnUse':
+        return 'Account bound on use';
+      case 'AccountBound':
+        return 'Account bound';
+      case 'Attuned':
+        return 'Attuned';
+      case 'BulkConsume':
+        return 'Bulk consumable';
+      case 'DeleteWarning':
+        return 'Warning on delete';
+      case 'HideSuffix':
+        return 'Suffix hidden';
+      case 'Infused':
+        return 'Infused';
+      case 'MonsterOnly':
+        return 'Monster only item';
+      case 'NoMysticForge':
+        return 'Cannot be used in the Mystic Forge';
+      case 'NoSalvage':
+        return 'Cannot be salvaged';
+      case 'NoSell':
+        return 'Cannot be sold';
+      case 'NotUpgradeable':
+        return 'Not upgradeable';
+      case 'NoUnderwater':
+        return 'Cannot be used underwater';
+      case 'SoulbindOnAcquire':
+        return 'Soul bound on acquire';
+      case 'SoulBindOnUse':
+        return 'Soul bound on use';
+      case 'Tonic':
+        return 'Tonic';
+      case 'Unique':
+        return 'Unique';
+      // this is a non-API flag
+      case 'Soulbound':
+        return 'Soulbound';
       default:
         return type;
     }
   }
+
+  List<String> getFilteredFlags(List<String> originalFlags) {
+    // Return an empty list if the flags are null. Now you only have to check
+    // if there are more than zero flags to determine whether or not to run the
+    // _buildFlags function.
+    if (originalFlags == null) return [];
+
+    List<String> filteredFlags = originalFlags.where((f) {
+      switch (section) {
+        case ItemSection.equipment:
+          if (f == 'HideSuffix') return false;
+          return true;
+        case ItemSection.bank:
+          if (f == 'NoUnderwater' || f == 'DeleteWarning' || f == 'HideSuffix') return false;
+          return true;
+        case ItemSection.inventory:
+          if (f == 'HideSuffix') return false;
+          return true;
+        case ItemSection.material:
+          return f != 'NoUnderwater'
+              && f != 'HideSuffix';
+        case ItemSection.tradingpost:
+        default: return f != 'HideSuffix';
+      }
+    }).toList();
+
+    // modify flags based on existence (or not) of another in the filtered list
+    if (section == ItemSection.equipment) {
+      if (filteredFlags.contains('AccountBound')) {
+        filteredFlags.removeWhere((element) => element == 'AccountBindOnUse');
```

Comment:
I'd use the `remove` function instead of `removeWhere` as it should perform a bit better.
So like this:
```
filteredFlags.remove('AccountBindOnUse');
```

Reply:
Yes, this makes more sense given the nature of the list. Updated.